services/queue/src/utils.js

```diff
@@ -37,17 +37,22 @@ const artifactUtils = {
    * This method will remove both the artifact in the db and underlying artifact.
    * But the artifact in the db will not be deleted if there is an error
    * deleting the underlying artifact.
+   *
+   * Not all S3-compatible storage providers support bulk delete, so we
+   * need to handle that case.
    */
-  async expire({ db, publicBucket, privateBucket, ignoreError, monitor, expires }) {
+  async expire({ db, publicBucket, privateBucket, ignoreError, monitor,
+    expires, useBulkDelete = false }) {
     let count = 0;
+    const maxPageSize = useBulkDelete ? 1000 : 100;
```

Comment:
maybe this requires some explanation .. 
bulk delete operation can support up to `1000` records, where in single removal we don't want to fire too many parallel requests, so we use `100` which was used before bulk feature was added.

Reply:
What do you think about making both of these values configurable as well? Maybe single deletes happen faster on GCS and we want to bump up that `100` value without having to make a code change?

Reply:
> maybe this requires some explanation ..

Can you add this explanation to the code as a comment too?

Reply:
yes! just added, thanks